test/unit/Cardano/Wallet/Primitive/AddressDiscoverySpec.hs

```diff
@@ -129,9 +140,16 @@ spec = do
             fromText @AddressPoolGap "20eiei" === Left (TextDecodingError err)
         it "fail fromText @AddressPoolGap \"raczej nie\"" $
             fromText @AddressPoolGap "raczej nie" === Left (TextDecodingError err)
-        where
-            err :: String
-            err = "An address pool gap must be a natural number between 10 and 100"
+
+    describe "PendingIxs & AddressScheme" $ do
+        it "Can always generate exactly `gap` different change addresses"
+            (property prop_nextChangeAddressGap)
+        it "After `gap` change addresses, the same one are yield in reverse order"
+            (property prop_changeAddressRotation)
+        it "Can generate new change addresses after discovering a pending one"
```

Comment:
I can't wrap my head around this property. So after pending address is discovered we should be able to generate a new change address.

In other words - it shouldn't be the case that we have a pending address and we are unable to generate a new change address. What would be the constrain to generate a new change address - is it only a pool gap ?

Reply:
> it shouldn't be the case that we have a pending address and we are unable to generate a new change address

It's actually the case. For the internal chain, we are actually free to do whatever we want. However, we do want to keep all information we need onto the blockchain (to not impede the wallet's portability) so, we've chosen to also track change addresses using an address pool gap (which is a choice consistent with Yoroi...). As a consequence, we can't generate change addresses outside of the gap. Still, when we discover an address that was pending, we now that the pool has extended and therefore, there are new addresses that we can now generate.